### PR TITLE
feat(vertexai): support system instruction

### PR DESCRIPTION
**Pull request by glaforge**
## Description

Fixes b/338204721 
(internal bug reference)

Add support for system instruction.

### CODE REVIEW

**Review by gericdong — APPROVED** [review]

File: vertexai/snippets/src/main/java/vertexai/gemini/WithSystemInstruction.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-google-cloud-project-id";
+    String location = "us-central1";
+    String modelName = "gemini-1.5-pro-preview-0409";
+
+    String output = translateToFrench(projectId, location, modelName);
+    System.out.println(output);
+  }
+
+  // Ask the model to translate from English to French thanks to a system instruction.
```
Comment (Sita04) [inline vertexai/snippets/src/main/java/vertexai/gemini/WithSystemInstruction.java:38]:
```suggestion
  // Ask the model to translate from English to French with a system instruction.
```

**Review by Sita04 — APPROVED** [review]
LGTM. Thanks, just one nit in the comment.

File: vertexai/snippets/src/main/java/vertexai/gemini/WithSystemInstruction.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-google-cloud-project-id";
+    String location = "us-central1";
+    String modelName = "gemini-1.5-pro-preview-0409";
+
+    String output = translateToFrench(projectId, location, modelName);
+    System.out.println(output);
+  }
```
Comment (minherz) [inline vertexai/snippets/src/main/java/vertexai/gemini/WithSystemInstruction.java:36]: nit: It is a good practice to exclude the code of `main()` from the code snippet. To do this you need to place _TWO_ region tag START/END comments around necessary imports and around the code sample method.

File: vertexai/snippets/src/main/java/vertexai/gemini/WithSystemInstruction.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      String output;
+
+      GenerativeModel model = new GenerativeModel(modelName, vertexAI)
+          .withSystemInstruction(ContentMaker.fromString("You are a helpful assistant.\n"
+            + "Your mission is to translate text in English to French."));
+
+      GenerateContentResponse response = model.generateContent("User input: I like bagels.\n"
+          + "Answer:");
+      output = ResponseHandler.getText(response);
+      return output;
```
Comment (minherz) [inline vertexai/snippets/src/main/java/vertexai/gemini/WithSystemInstruction.java:53]: nit: I think it can be a direct return.

**Review by minherz — APPROVED** [review]

**Pull request merged**